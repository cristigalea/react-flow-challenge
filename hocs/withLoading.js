/* @flow */
import * as React from 'react';

type Props = {
    loading?: boolean,
    error?: ?Error
};

const withLoading = (Component: any) => ({loading, error, ...props}: Props) => {
    if (loading) {
        return <div>Loading...</div>;
    }
    if (error) {
        return <div>Error!</div>;
    }
    return <Component {...props} />;
};

export default withLoading;
