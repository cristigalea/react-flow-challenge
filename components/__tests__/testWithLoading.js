/* @noflow */
import React from 'react';
import {render} from 'enzyme';

import withLoading from '../../hocs/withLoading';

describe('<DescriptionWithLinks />', () => {
    const PassedComponent = ({someProp, otherProp}) => <h3>Passed values: {someProp}, {otherProp}</h3>;
    const Component = withLoading(PassedComponent);

    it('should render the error message when the error flag is true', () => {
        const wrapper = render(<Component error={true} someProp="test" />);
        expect(wrapper.text()).toBe('Error!');
        expect(wrapper).toMatchSnapshot();
    });
    it('should render the loading message when the loading flag is true', () => {
        const wrapper = render(<Component loading={true} someProp="test" />);
        expect(wrapper.text()).toBe('Loading...');
        expect(wrapper).toMatchSnapshot();
    });
    it('should render the passed component when the error & loading flags are not true', () => {
        const wrapper = render(<Component someProp="test" otherProp="test2" />);
        expect(wrapper.text()).toBe('Passed values: test, test2');
        expect(wrapper).toMatchSnapshot();
    });
});
