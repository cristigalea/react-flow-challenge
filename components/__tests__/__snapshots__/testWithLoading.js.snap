// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DescriptionWithLinks /> should render the error message when the error flag is true 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Error!",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;

exports[`<DescriptionWithLinks /> should render the loading message when the loading flag is true 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Loading...",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "div",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;

exports[`<DescriptionWithLinks /> should render the passed component when the error & loading flags are not true 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "data": "Passed values: test, test2",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "h3",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
